product_abc_classification/models/abc_classification_level.py

```diff
@@ -8,28 +8,71 @@
 class ABCClassificationProfileLevel(models.Model):
     _name = "abc.classification.profile.level"
     _description = "ABC Classification Profile Level"
-    _order = "percentage desc, id desc"
+    _order = "sequence desc"
 
-    percentage = fields.Float(default=0.0, required=True, string="%")
+    classification_type = fields.Selection(
+        related="profile_id.classification_type", string="Classification Type"
+    )
+    percentage = fields.Float(default=0.0, string="%")
+    fixed = fields.Monetary(default=0.0, string="Fixed Value")
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        related="profile_id.company_id",
+        store=True,
+    )
+    currency_id = fields.Many2one(
+        "res.currency", string="Currency", store=True, related="company_id.currency_id"
+    )
     profile_id = fields.Many2one("abc.classification.profile")
+    sequence = fields.Integer(
+        string="Sequence", compute="_compute_sequence", store=True
+    )
 
     def name_get(self):
         def _get_sort_key_percentage(rec):
             return rec.percentage
 
+        def _get_sort_key_fixed_price(rec):
+            return rec.fixed
+
+        method_dict = {
+            "percentage": _get_sort_key_percentage,
+            "fixed": _get_sort_key_fixed_price,
+        }
         res = []
         for profile in self.mapped("profile_id"):
+            classification_type = profile.classification_type
             for i, level in enumerate(
-                profile.level_ids.sorted(key=_get_sort_key_percentage, reverse=True)
+                profile.level_ids.sorted(
+                    key=method_dict[classification_type], reverse=True
+                )
             ):
-                name = "{} ({}%)".format(chr(65 + i), level.percentage)
+                name = "{} ({}".format(chr(65 + i), level[classification_type])
+                if classification_type == "percentage":
+                    name += "%)"
```

Comment:
Maybe we want to end the string with another character, like the currency symbol. A universal solution would be to have the proper char computed depending on the type